### PR TITLE
Update with eoan's gir-files

### PR DESCRIPTION
**Pull request by EPashkin**
Application of https://github.com/gtk-rs/gir-files/pull/37

IMHO this better wait until we finish current release

### CODE REVIEW

**Comment by EPashkin** [comment]
At minimum, this diff caused a build error, so Gir needs to be fixed
```diff
#[repr(C)]
#[derive(Copy, Clone)]
pub struct GTestLogMsg {
    pub log_type: GTestLogType,
    pub n_strings: c_uint,
    pub strings: *mut *mut c_char,
    pub n_nums: c_uint,
-    pub nums: *mut c_long,
+    _truncated_record_marker: c_void,
+    // /*Unimplemented*/*mut long double
}
```
from this GLib-2.0.gir change
```diff
@@ -23571,11 +24778,12 @@ zero then @fixture will be equal to @user_data.</doc>
      <field name="n_nums" writable="1">
        <type name="guint" c:type="guint"/>
      </field>
-      <field name="nums" writable="1">
-        <type name="glong" c:type="long*"/>
+      <field name="nums" introspectable="0" writable="1">
+        <type name="long double" c:type="long double*"/>
      </field>
```

**Comment by GuillaumeGomez** [comment]
Seems like `gdk-sys` was not correctly generated.

**Comment by EPashkin** [comment]
@GuillaumeGomez Why do you think so?
Almost all changes in Gdk-3.0.gir are `<doc xml:space="preserve" filename="gdk/gdkapplaunchcontext.c" line="236">` or `<source-position filename="gdk/gdkproperty.h" line="59"/>`

**Comment by GuillaumeGomez** [comment]
The CI. :)

**Comment by EPashkin** [comment]
Is normal that regen_check fail, PR don't have gir/gir-files changes

**Comment by EPashkin** [comment]
Merged gir's PR and updated this

**Comment by sdroege** [comment]
Looks good to me, now only needs regens and enabling of all the new API in the other crates

**Comment by EPashkin** [comment]
I can't start "regens and enabling of all the new API in the other crates" until this merged 😉

**Comment by GuillaumeGomez** [comment]
Thanks!

**Comment by sdroege** [comment]
> I can't start "regens and enabling of all the new API in the other crates" until this merged wink

Sure, I meant once this is merged :) You take care of that?

File: gtk-sys/src/lib.rs [hunk 1]
```diff
@@ -1305,7 +1305,7 @@ impl ::std::fmt::Debug for GtkBindingArg_d {
 #[repr(C)]
 #[derive(Copy, Clone)]
 pub union GtkTextAppearance_u1 {
-    pub rgba: [*mut gdk::GdkRGBA; 2],
+    pub rgba: *mut [*mut gdk::GdkRGBA; 2],
```
Comment (sdroege) [inline gtk-sys/src/lib.rs]: This was wrong

File: gtk-sys/src/lib.rs [hunk 2]
```diff
@@ -13545,7 +13568,7 @@ impl ::std::fmt::Debug for GtkRangeAccessible {
 pub struct GtkRcStyle {
     pub parent_instance: gobject::GObject,
     pub name: *mut c_char,
-    pub bg_pixmap_name: [*mut c_char; 5],
+    pub bg_pixmap_name: *mut [*mut c_char; 5],
```
Comment (sdroege) [inline gtk-sys/src/lib.rs]: And this

File: gtk-sys/src/lib.rs [hunk 3]
```diff
@@ -14162,8 +14186,14 @@ pub struct GtkStyle {
     pub font_desc: *mut pango::PangoFontDescription,
     pub xthickness: c_int,
     pub ythickness: c_int,
-    _truncated_record_marker: c_void,
-    // /*Ignored*/field background has incomplete type
+    pub background: *mut [*mut cairo::cairo_pattern_t; 5],
```
Comment (sdroege) [inline gtk-sys/src/lib.rs]: This too

**Pull request merged**